config/main.py

```diff
@@ -1712,6 +1713,10 @@ def load_minigraph(db, no_service_restart):
     if os.path.isfile(DEFAULT_GOLDEN_CONFIG_DB_FILE):
         override_config_by(DEFAULT_GOLDEN_CONFIG_DB_FILE)
 
+    # Keep device in maintenance with TSA 
+    if traffic_shift_away:
+        clicommon.run_command("TSA", display_cmd=True)
```

Comment:
>run_command [](http://example.com/codeflow?start=18&length=11)

This may override above golden config. Actually you can achieve your goal with golden config, if golden config is used.

Reply:
This will only apply when the command is triggered with "--traffic_shift_away" option, in which case, I think the user intention is to override golden config and keep the device in maintenance.

Reply:
Your implementation on a special case that DEFAULT_GOLDEN_CONFIG_DB_FILE exists and user provided "--traffic_shift_away" option:
1. first override by golden config
2. TSA applied

So TSA will win the override because it is applied later. However in this special case, I think golden config should be the winner since user could use golden config to achieve the effect of TSA. Suggest ignore TSA option if DEFAULT_GOLDEN_CONFIG_DB_FILE exists and user provided "--traffic_shift_away" option

Reply:
> Your implementation on a special case that DEFAULT_GOLDEN_CONFIG_DB_FILE exists and user provided "--traffic_shift_away" option:
> 
> 1. first override by golden config
> 2. TSA applied
> 
> So TSA will win the override because it is applied later. However in this special case, I think golden config should be the winner since user could use golden config to achieve the effect of TSA. Suggest ~ignore TSA option if DEFAULT_GOLDEN_CONFIG_DB_FILE exists and user provided "--traffic_shift_away" option~ swap the order of Step 1 and Step 2

That's right. Customers can achieve TSA status by modifying the golden config.

Reply:
Thanks, updated and added a warning since the CLI option may not work in this case anymore